pkg/config/config.go

```diff
@@ -48,3 +51,48 @@ func NewKVConfiguration(fileName string) (KVConfiguration, error) {
 	err = decoder.Decode(&configuration)
 	return configuration, err
 }
+
+func (c *KVConfiguration) GetReplications() []string {
+	var localStores []string
+	var neighborStores []string
+	var remoteStores []string
+	for _, store := range c.Stores {
+		switch region := store.RegionType; region {
+		case "local":
+			localStores = append(localStores, fmt.Sprintf("%s:%d", store.Host, store.Port))
+		case "neighbor":
+			neighborStores = append(neighborStores, fmt.Sprintf("%s:%d", store.Host, store.Port))
+		case "remote":
+			remoteStores = append(remoteStores, fmt.Sprintf("%s:%d", store.Host, store.Port))
+		}
+	}
+	n := len(localStores)
+	replications := make([]string, n)
+	for idx := 0; idx < n; idx++ {
+		total := c.ReplicaNum
+		replics := make([]string, total)
+		total--
+		if remote := selectRandom(remoteStores); remote != "" {
+			replics[total] = remote
+			total--
+		}
+		if neighbor := selectRandom(neighborStores); neighbor != "" {
+			replics[total] = neighbor
+			total--
+		}
+		for i := 0; i < total+1; i++ {
+			replics[total] = localStores[(idx+i)%n]
+			total--
+		}
+		replications[idx] = strings.Join(replics, ",")
+	}
+	return replications
+}
+
+func selectRandom(array []string) string {
+	if len(array) == 0 {
+		return ""
+	}
+	randomIndex := rand.Intn(len(array))
```

Comment:
has rand been seeded?

Reply:
Please refer to https://github.com/golang/go/blob/master/src/math/rand/rng.go#L25
If a rand is seeded, it will give a deterministic sequence of pseudo-random numbers. We use rand for now and later it will be replaced by picking by loading information.

Reply:
Just one comment: a common programming practice to avoid deterministic sequences is to add system time to a seed, e.g. rand.Seed(time.Now().UnixNano()). Since later on we may need other approaches, this is just for your reference.

Reply:
I added. But later it will be replaced by store load information :)